### ISSUE TITLE
I have been testing this repo and...

### ISSUE THREAD

**Issue by davidmartinrius** [Anvarjon/Age-Gender-Classification#3]
Hello @Anvarjon 

I have been testing this repo with multiple unseen audio sample files, with different ages and genders.

It is very imprecise. Sometimes men's voices are recognized as women's voices and vice versa. As well as voices of older people, it  identifies them as adolescents.

I guess it is not working with unseen samples, but only samples inside the dataset.

I really would like to integrate it to the [speech dataset generator](https://github.com/davidmartinrius/speech-dataset-generator)

But, I would like to always use it with unseen samples. Do you think there could be any improvement to this repository to achieve this?

A finetuning is not an option, I need something that works always with unseen samples

**Comment by Anvarjon** [comment]
Hello @davidmartinrius,

Thank you for your interest in this project. I wanted to let you know that the pre-trained models were not trained on cross-corpus datasets. As a result, the model performance may be poor when testing on audio samples that it has not been trained on.

To improve model performance, there are two ways to proceed. The first way is to collect more data samples from other sources and train a model with it. The second way is to make changes to the model architecture, increase the input size, and train the model with different training parameters to find the best possible hyperparameters and model architecture.

I can try the second way to improve the model accuracy. It would be helpful if you could share some of your testing audio samples with me. This way, I could test the model after improving its accuracy. 

Thank you!

**Comment by davidmartinrius** [comment]
Either the model architecture is not well defined or the model training results are not good. In any case, it should be possible to use it on unseen samples, otherwise it is not functional in the real world.

I hope you achieve good results! For now I have ruled out integrating it into the speech-dataset-generator.
